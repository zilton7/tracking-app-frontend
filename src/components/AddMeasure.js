import React, { useState } from "react";
import { useSelector } from "react-redux";

const AddMeasure = () => {
  const [data, setData] = useState({
    measurementId: null,
    measureData: null,
  });

  // Load measurements from state
  let measurements = useSelector((state) => state.measurements.measurements);

  const handleClick = () => {
    const values = "calculate(buttonName, data)";
    setData({
      measurementId: "calculation.total",
      measureData: "calculation.next",
    });
  };

  return (
    <div className="add-measure">
      <select name="select-measurements" id="select-measurements">
        {measurements.map((measurement) => (
          <option value={measurement.id}>{measurement.name}</option>
        ))}
      </select>
      <input placeholder="55.5" />
    </div>
  );
};

export default AddMeasure;
