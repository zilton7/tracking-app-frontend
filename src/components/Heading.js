import { useSelector } from "react-redux";

const Heading = () => {
  // Load username from state
  let username = useSelector((state) => state.user.username);

  return <div className="heading">{username}'s Tracking App</div>;
};

export default Heading;
