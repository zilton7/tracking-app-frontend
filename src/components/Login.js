import { useDispatch } from "react-redux";
import login from "../actions/loginAction";

const Login = () => {
  const dispatch = useDispatch();

  const handleClick = () => {
    const username = document.getElementById("username").value;
    if (username != "") {
      dispatch(login(username));
    } else {
      alert("Username can not be empty");
    }
  };

  return (
    <div className="login-page">
      <div className="container">
        <h1>Tracking App</h1>
        <input id="username" placeholder="Enter Your Name" />
        <button onClick={handleClick}>Login</button>
      </div>
    </div>
  );
};

export default Login;
