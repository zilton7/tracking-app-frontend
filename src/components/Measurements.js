import React, { useEffect } from "react";
import axios from "axios";
import MeasurementItem from "./MeasurementItem";
import bicepLeft from "../images/bicep-left.png";
import bicepRight from "../images/bicep-right.png";
import waist from "../images/waist.png";
import hip from "../images/hip.png";
import legLeft from "../images/leg-left.png";
import legRight from "../images/leg-right.png";

const Measurements = () => {
  function checkLoginStatus() {
    axios
      .get("https://tracking-app-be-zil.herokuapp.com/api/v1/measurements")
      .then((response) => {
        console.log(response.data);
      })
      .catch((error) => error);
  }
  useEffect(() => {
    checkLoginStatus();
  });

  return (
    <div className="measurements">
      <div className="measurements-left">
        <MeasurementItem image={bicepLeft} name="Left Bicep" measure="50.6" />
        <MeasurementItem image={waist} name="Waist" measure="50.6" />
        <MeasurementItem image={legLeft} name="Left Leg" measure="50.6" />
      </div>
      <div className="measurements-right">
        <MeasurementItem image={bicepRight} name="Right Bicep" measure="50.6" />
        <MeasurementItem image={hip} name="Hip" measure="50.6" />
        <MeasurementItem image={legRight} name="Right Leg" measure="50.6" />
      </div>
    </div>
  );
};

export default Measurements;
