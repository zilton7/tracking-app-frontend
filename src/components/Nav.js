const Nav = () => (
  <nav>
    <div class="bottom-buttons">
      <div class="button-wrapper">
        <i class="far fa-plus-square"></i>
        <p>Add measure</p>
      </div>
      <div class="button-wrapper">
        <i class="fas fa-chart-line"></i>
        <p>Tracker</p>
      </div>
      <div class="button-wrapper">
        <i class="fas fa-chart-pie"></i>
        <p>Progress</p>
      </div>
      <div class="button-wrapper">
        <i class="fas fa-ellipsis-h"></i>
        <p>More</p>
      </div>
    </div>
  </nav>
);

export default Nav;
