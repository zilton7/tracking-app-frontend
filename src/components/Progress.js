import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import axios from "axios";
import ProgressItem from "./ProgressItem";

const Progress = () => {
  let { measurementId } = useParams();
  const [measures, setMeasures] = useState([]);
  const [measureName, setMeasureName] = useState([]);

  useEffect(() => {
    axios
      .get(
        `https://tracking-app-be-zil.herokuapp.com/api/v1/measurements/${measurementId}`
      )
      .then((response) => {
        setMeasures(response.data.measures);
        setMeasureName(response.data.name);
      })
      .catch((err) => err);
  }, []);

  return (
    <div className="progress-container">
      <h3>Your {measureName} Progress</h3>
      {measures.map((measure) => (
        <ProgressItem date={measure.created_at} data={measure.data} />
      ))}
    </div>
  );
};

export default Progress;
