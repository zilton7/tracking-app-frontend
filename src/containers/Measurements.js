import React from 'react';
import { useSelector } from 'react-redux';
import MeasurementItem from '../components/MeasurementItem';

const Measurements = () => {
  // Load measurements from state
  const measurements = useSelector((state) => state.measurements.measurements);

  return (
    <div className="measurements">
      {measurements.map((measurement) => (
        <MeasurementItem
          key={measurement.id}
          id={measurement.id}
          image={measurement.image}
          name={measurement.name}
        />
      ))}
    </div>
  );
};

export default Measurements;
