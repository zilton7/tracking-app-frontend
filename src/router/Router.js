import React from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import Measurements from "../containers/Measurements";
import Progress from "../containers/Progress";
import AddMeasure from "../components/AddMeasure";
import Nav from "../components/Nav";

function Router() {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/" component={Measurements} />
        <Route exact path="/add" component={AddMeasure} />
        <Route exact path="/tracker" component={Measurements} />
        <Route path="/progress/:measurementId" component={Progress} />
      </Switch>
      <Nav />
    </BrowserRouter>
  );
}

export default Router;
