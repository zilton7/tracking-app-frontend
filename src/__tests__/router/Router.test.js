import React from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import Router from "../../router/Router";
import store from "../../store/index";
import Login from "../../components/Login";
import Heading from "../../components/Heading";
import Measurements from "../../containers/Measurements";
import AddMeasure from "../../components/AddMeasure";
import Progress from "../../containers/Progress";
import Nav from "../../components/Nav";

afterEach(cleanup);

const user = {
  loggedIn: true,
};

it("Takes website's snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <div className="App">
        {user.loggedIn ? (
          <BrowserRouter>
            <Heading />
            <Switch>
              <Route exact path="/" component={Measurements} />
              <Route exact path="/add" component={AddMeasure} />
              <Route exact path="/tracker" component={Measurements} />
              <Route path="/progress/:measurementId" component={Progress} />
            </Switch>
            <Nav />
          </BrowserRouter>
        ) : (
          <Login />
        )}
      </div>
    </Provider>
  );
  expect(
    asFragment(
      <Provider store={store}>
        <BrowserRouter>
          <Heading />
          <Switch>
            <Route exact path="/" component={Measurements} />
            <Route exact path="/add" component={AddMeasure} />
            <Route exact path="/tracker" component={Measurements} />
            <Route path="/progress/:measurementId" component={Progress} />
          </Switch>
          <Nav />
        </BrowserRouter>
      </Provider>
    )
  ).toMatchSnapshot();
});

it("Displays app's title", () => {
  render(
    <Provider store={store}>
      <div className="App">
        {user.loggedIn ? (
          <BrowserRouter>
            <Heading />
            <Switch>
              <Route exact path="/" component={Measurements} />
              <Route exact path="/add" component={AddMeasure} />
              <Route exact path="/tracker" component={Measurements} />
              <Route path="/progress/:measurementId" component={Progress} />
            </Switch>
            <Nav />
          </BrowserRouter>
        ) : (
          <Login />
        )}
      </div>
    </Provider>
  );
  const element = screen.getByText(/Tracking App/i);
  expect(element).toBeInTheDocument();
});

// it("Displays nav bar", () => {
//   render(
//     <Provider store={store}>
//       <BrowserRouter>
//         <App />
//       </BrowserRouter>
//     </Provider>
//   );
//   const element = screen.getByPlaceholderText(/Search for Meal/i);
//   expect(element).toBeInTheDocument();
// });
