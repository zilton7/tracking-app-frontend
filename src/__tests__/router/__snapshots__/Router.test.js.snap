// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes website's snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="heading"
    >
      's Tracking App
    </div>
    <div
      class="measurements"
    />
    <nav>
      <div
        class="bottom-buttons"
      >
        <a
          class="button-wrapper"
          href="/add"
        >
          <i
            class="far fa-plus-square"
          />
          <p>
            Add measure
          </p>
        </a>
        <a
          class="button-wrapper"
          href="/tracker"
        >
          <i
            class="fas fa-chart-line"
          />
          <p>
            Tracker
          </p>
        </a>
        <a
          class="button-wrapper"
          href="/progress/1"
        >
          <i
            class="fas fa-chart-pie"
          />
          <p>
            Progress
          </p>
        </a>
        <div
          class="button-wrapper"
        >
          <i
            class="fas fa-sign-out-alt"
          />
          <p>
            Logout
          </p>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
